Demo/Common/IntegrationExamples/Components/CardComponentExample.swift

```diff
@@ -0,0 +1,200 @@
+        handler.configuration.threeDS.delegateAuthentication = ConfigurationConstants.delegatedAuthenticationConfigurations
+        handler.configuration.threeDS.requestorAppURL = URL(string: ConfigurationConstants.returnUrl)
+        handler.delegate = self
+        handler.presentationDelegate = self
+        return handler
+    }()
+
+    // MARK: - Initializers
+
+    internal init() {}
+
+    // MARK: - Networking
+
+    internal func requestInitialData(completion: ((PaymentMethods?, Error?) -> Void)?) {
+        requestPaymentMethods(order: nil) { [weak self] paymentMethods, errorResponse in
+            guard paymentMethods != nil else {
+                guard let errorResponse = errorResponse else {
+                    return
+                }
+                self?.presenter?.presentAlert(with: errorResponse, retryHandler: {
+                    self?.requestPaymentMethods(order: nil, completion: completion)
+                })
+                return
+            }
+            self?.paymentMethods = paymentMethods
+        }
+    }
+
+    internal func presentCardComponent() {
+        guard let component = cardComponent(from: paymentMethods) else { return }
+        component.cardComponentDelegate = self
+        present(component, delegate: self)
+    }
+
+    internal func cardComponent(from paymentMethods: PaymentMethods?) -> CardComponent? {
+        guard let paymentMethods = paymentMethods,
+              let paymentMethod = paymentMethods.paymentMethod(ofType: CardPaymentMethod.self) else { return nil }
+        let style = FormComponentStyle()
+        let config = CardComponent.Configuration(style: style)
+        return CardComponent(paymentMethod: paymentMethod,
+                             context: context,
+                             configuration: config)
+    }
+
+    // MARK: - Presentation
+
+    private func present(_ component: PresentableComponent, delegate: (PaymentComponentDelegate & ActionComponentDelegate)) {
+        if let paymentComponent = component as? PaymentComponent {
+            paymentComponent.delegate = delegate
+        }
+
+        if let actionComponent = component as? ActionComponent {
+            actionComponent.delegate = delegate
+        }
+
+        cardComponent = component
+        guard component.requiresModalPresentation else {
+            presenter?.present(viewController: component.viewController, completion: nil)
+            return
+        }
+
+        let navigation = UINavigationController(rootViewController: component.viewController)
+        component.viewController.navigationItem.rightBarButtonItem = .init(barButtonSystemItem: .cancel,
+                                                                           target: self,
+                                                                           action: #selector(cancelPressed))
+        presenter?.present(viewController: navigation, completion: nil)
+    }
```

Comment:
we can move `cardComponent = component` to `presentCardComponent()`. Then we can have:

```suggestion
    private func present(component: PresentableComponent) {
        let navigation = UINavigationController(rootViewController: component.viewController)
        component.viewController.navigationItem.rightBarButtonItem = .init(barButtonSystemItem: .cancel,
                                                                           target: self,
                                                                           action: #selector(cancelPressed))
        presenter?.present(viewController: navigation, completion: nil)
    }
```

Reply:
I have made this change. e7c03971d28f7f4fb10a23fe39580f2dacb88ac4